skopt/space/space.py

```diff
@@ -273,10 +276,18 @@ def __init__(self, categories, prior=None):
         * `prior` [list, shape=(categories,), default=None]:
             Prior probabilities for each category. By default all categories
             are equally likely.
+
+        * `transform` [string, "onehot" or "identity"]:
+            When `transform` is set to "identity", the transformed space
+            is the same as the original space.
         """
         self.categories = categories
-        self.transformer = CategoricalEncoder()
-        self.transformer.fit(self.categories)
+        self.transform_ = transform
```

Comment:
Not sure I like setting both `self.transform` and `self.transform_`. This is confusing. Only `self.transform` seems necessary.